apps/balancer-tools/src/app/apr/api/route.ts

```diff
@@ -109,10 +110,13 @@ export async function GET(request: NextRequest) {
     return NextResponse.json(
       await fetchDataForPoolIdDateRange(poolId, startAt, endAt),
     );
-  } else if (poolId) {
-    responseData = await fetchDataForPoolId(poolId);
   } else if (startAt && endAt) {
-    responseData = await fetchDataForDateRange(startAt, endAt, parsedParams);
+    const cachedData = withCache(
+      (...args: [typeof startAt, typeof endAt, typeof parsedParams]) =>
```

Comment:
Think I found the root cause, it's because we are passing the params as args. Therefore, with different args different cache